### PR TITLE
fix(web-components): no margin on single icon in buttons

### PR DESCRIPTION
**Pull request by evenstensberg**
## Summary of the changes
#2178

## Related issue
Tell us the issue number. If suggesting an improvement or component, please discuss it with us in an issue first.

## Todo Even

- Update tests
- Only icon with no variant prefix needs correct margins
- Check that the code is where it is supposed to be

### General 

- [ ] Changes to docs package checked and committed.
- [ ] All acceptance criteria reviewed and met. 

### Testing

- [ ] Relevant unit tests and visual regression tests added. 
- [ ] Visual testing against Figma component specification completed. 
- [ ] Playground stories in React Storybook up to date, with any prop changes and additions addressed. 

### Accessibility 

- [ ] Accessibility Insights FastPass performed.
- [ ] A11y unit test added and yields no issues.
- [ ] A11y plug-in on Storybook yields no issues. 
- [ ] Manual screen reader testing performed using NVDA and VoiceOver. 
- [ ] Manual keyboard testing for keyboard controls and logical focus order. 
- [ ] Correct roles used and ARIA attributes used correctly where required. 
- [ ] Logical heading structure is maintained, and the HTML elements used for headings can be changed to fit within the wider page structure. 

### Resize/zoom behaviour 

- [ ] Page can be zoomed to 400% with no loss of content. 
- [ ] Screen magnifier used with no issues. 
- [ ] Text resized to 200% with no loss of content.
- [ ] Text spacing increased as per the [WCAG 1.4.12 success criterion](https://www.w3.org/TR/WCAG21/#text-spacing) with no loss of content.

### System modes

- [ ] Browser setting 'prefers reduced motion' tested. No animations or motion visible whilst this setting is on. 
- [ ] Windows High Contrast mode tested with no loss of content.
- [ ] System light and dark mode tested with no loss of content.
- [ ] Browser support tested (Chrome, Safari, Firefox and Edge). 

### Testing content extremes

- [ ] Min/max content examples tested with no loss of content or overflow. 
- [ ] All prop combinations work without issue. 
- [ ] Tested for FOUC (Flash of Unstyled Content) in both SSR (Server-Side Rendering) and SSG (Static Site Generation) settings.
- [ ] Controlled and uncontrolled input components tested.

### CODE REVIEW

**Comment by evenstensberg** [comment]
If the code changes is ok, I'll update the tests

File: packages/web-components/src/components/ic-button/ic-button.css [hunk 1]
```diff
@@ -515,7 +515,6 @@
   height: var(--height, var(--ic-space-xl));
   width: var(--ic-space-xl);
   padding: 0.375rem;
-  margin: 0;
```
Comment (evenstensberg) [inline packages/web-components/src/components/ic-button/ic-button.css]: Maybe the margin removals should be in this selector

**Comment by GCHQ-Developer-530** [comment]
> If the code changes is ok, I'll update the tests

The code changes you've made look good to me. The commits will need squashing into one web-components commit, but I think you should be good to go ahead and update the tests.

**Comment by evenstensberg** [comment]
@GCHQ-Developer-530 I've updated the test I can fix. Need to remove the baseline images that fail, you will need to tell me which baseline images to remove. Also test this thoroughly, so you dont ship a broken pr. The applied changes is affecting a lot of components it seems from the ci run

**Comment by GCHQ-Developer-530** [comment]
Hi @evenstensberg, sorry actually looking into it I found a member of our team had started this work before and it looks like it should be as simple as just removing the margin in a few places. See the [closed PR button css file](https://github.com/mi6/ic-ui-kit/pull/2174/files#diff-432d0bad05e99bea6d63719529d49829a3db8006dcafd7f028758754078b1c7b). The spec tests then shouldn't need updating. The work in the tsx file to add a new class name should also not be needed. Then there should just be some work for cypress tests. Please could you take a look at that PR and make updates to this PR accordingly? Thank you

**Comment by evenstensberg** [comment]
@GCHQ-Developer-530 Rebased and updated. Need to know the baseline images to remove. Thanks for reviewing!

**Comment by GCHQ-Developer-530** [comment]
There's quite a lot of snapshots to be removed this time. Here's a list of the failing tests:

Button - should render icon buttons
Button - renders with icon variant button group (there seems to be a duplicate of this too so if you could delete that test from the `cy.tsx` file at the same time, that’d be great)
Card - should render interaction button
Card - should render with top image
Card - should render with middle image
Data Entity - should render with button in end component slot
Dialog - All tests failing - delete every dialog cypress image
Pagination - All tests failing - delete every pagination cypress image
Tab - should render responsive tabs
Tab - should render responsive tabs with light text
Tab - should render compact tab selector with horizontal scroll
Tab - should render min and max content tabs
Toast - should render multiline message
Toast - should render custom neutral icon
Toast - should render custom dismiss button aria label
Toggle button - should render icon variant
Top Navigation - should render icon buttons
Top Navigation - should render with long text values
Top Navigation - should render with long single-word app title
Top Navigation - should render with navigation items
Top Navigation - should render with slotted navigation items
Top Navigation - should render with navigation group
Top Navigation - should render with content centre aligned
Top Navigation - should render with content left aligned
Top Navigation - should render with different theme color
Top Navigation - Mobile - should render search
Top Navigation - Mobile - should render icon buttons
Top Navigation - Mobile - should render with long text values
Top Navigation - Mobile - should render with long single-word app title
Top Navigation - Mobile - should render with navigation items
Top Navigation - Mobile - should render with slotted navigation items
Top Navigation - Mobile - should render with navigation group
Top Navigation - Mobile - should render with menu open
Top Navigation - Mobile - should render with content centre aligned
Top Navigation - Mobile - should render with content left aligned
Top Navigation - Mobile - should render with different theme color

**Comment by evenstensberg** [comment]
## Wip

- [ ] Button - should render icon buttons
- [ ] Button - renders with icon variant button group (there seems to be a duplicate of this too so if you could delete that test from the cy.tsx file at the same time, that’d be great)
- [ ] Card - should render interaction button
- [ ] Card - should render with top image
- [ ] Card - should render with middle image
- [ ] Data Entity - should render with button in end component slot
- [ ] Dialog - All tests failing - delete every dialog cypress image
- [ ] Pagination - All tests failing - delete every pagination cypress image
- [ ] Tab - should render responsive tabs
- [ ] Tab - should render responsive tabs with light text
- [ ] Tab - should render compact tab selector with horizontal scroll
- [ ] Tab - should render min and max content tabs
- [ ] Toast - should render multiline message
- [ ] Toast - should render custom neutral icon
- [ ] Toast - should render custom dismiss button aria label
- [ ] Toggle button - should render icon variant
- [ ] Top Navigation - should render icon buttons
- [ ] Top Navigation - should render with long text values
- [ ] Top Navigation - should render with long single-word app title
- [ ] Top Navigation - should render with navigation items
- [ ] Top Navigation - should render with slotted navigation items
- [ ] Top Navigation - should render with navigation group
- [ ] Top Navigation - should render with content centre aligned
- [ ] Top Navigation - should render with content left aligned
- [ ] Top Navigation - should render with different theme color
- [ ] Top Navigation - Mobile - should render search
- [ ] Top Navigation - Mobile - should render icon buttons
- [ ] Top Navigation - Mobile - should render with long text values
- [ ] Top Navigation - Mobile - should render with long single-word app title
- [ ] Top Navigation - Mobile - should render with navigation items
- [ ] Top Navigation - Mobile - should render with slotted navigation items
- [ ] Top Navigation - Mobile - should render with navigation group
- [ ] Top Navigation - Mobile - should render with menu open
- [ ] Top Navigation - Mobile - should render with content centre aligned
- [ ] Top Navigation - Mobile - should render with content left aligned
- [ ] Top Navigation - Mobile - should render with different theme color

**Comment by evenstensberg** [comment]
@GCHQ-Developer-530 After removing this, I found out the syntax for the failing tests in the CI, so I won't be asking you more about the baseline images. In the future -> Whenever a PR is ok codewise, I will fix the failing tests without having to know which ones to remove. Thanks!

**Comment by GCHQ-Developer-530** [comment]
HI @evenstensberg, I had a go at running this locally to make sure it doesn't break the other components. The only issue I've noticed is that now the margin has been removed, the dismiss button on dialog looks a little odd. Especially when focused and the focus indicator is touching the top of the dialog. Could you maybe add a 1px top and bottom padding/margin specifically to the dismiss button in dialog to fix this?

**Comment by GCHQ-Developer-530** [comment]
Also please could the cypress changes be split out into a separate React scope commit? Thank you 😊

**Comment by evenstensberg** [comment]
Okay. Will fix this after https://github.com/mi6/ic-ui-kit/pull/2224 is done

**Comment by evenstensberg** [comment]
@GCHQ-Developer-530 Also, how many PRs do I need to land to get swag? 😅 Starting university in August, so I won't be contributing actively when that starts.

**Comment by GCHQ-Developer-530** [comment]
@evenstensberg we're still doing work behind the scenes to look into swag for contributors, so the timeline is a little unknown at the moment

**Comment by evenstensberg** [comment]
@GCHQ-Developer-530 Okay. I'm a webpack maintainer and we use to send stickers and t-shirts when we do swag. 

Some personal favourites:

- Pin
- Mug
- T-shirt
- Stickers

**Comment by evenstensberg** [comment]
> HI @evenstensberg, I had a go at running this locally to make sure it doesn't break the other components. The only issue I've noticed is that now the margin has been removed, the dismiss button on dialog looks a little odd. Especially when focused and the focus indicator is touching the top of the dialog. Could you maybe add a 1px top and bottom padding/margin specifically to the dismiss button in dialog to fix this? This should mean the dialog screenshots don't have to be removed as well.

@GCHQ-Developer-530 Could you give me a screenshot for the dialog buttons that are wrong? I'm running storybook locally and they seem ok to me.

**Comment by GCHQ-Developer-530** [comment]
> > HI @evenstensberg, I had a go at running this locally to make sure it doesn't break the other components. The only issue I've noticed is that now the margin has been removed, the dismiss button on dialog looks a little odd. Especially when focused and the focus indicator is touching the top of the dialog. Could you maybe add a 1px top and bottom padding/margin specifically to the dismiss button in dialog to fix this? This should mean the dialog screenshots don't have to be removed as well.
> 
> @GCHQ-Developer-530 Could you give me a screenshot for the dialog buttons that are wrong? I'm running storybook locally and they seem ok to me.

It's just that when focused the close icon on dialog has the focus indicator touching the top of the dialog but the side still doesn't due to padding, so it looked better with the 1px margin that was there before. Just for this button though, the other ones are fine.

**Pull request closed without merging**